### PR TITLE
Explain YAML string quoting, and lint examples

### LINKED ISSUE #322: Why are some strings quoted?
In the examples, some strings are quoted, and some aren't:

```
authors:
  - family-names: Druskat
    given-names: Stephan
    name-particle: von
    name-suffix: III
    address: "742 Evergreen Terrace"
    affiliation: "German Aerospace Center (DLR)"
    alias: "sdruskat"
    city: "Berlin"
    country: "DE"
    email: "sdruskat@research-project.org"
    fax: +12-3456-7890
    orcid: "https://orcid.org/0000-0003-4925-7248"
    post-code: 90210
    region: Renfrewshire
    tel: +12-345-6789098
    website: "https://research-project.org"
  ```
and

```
abstract: This is my awesome research software. It does many things.
authors:
  - family-names: Druskat
    given-names: Stephan
    orcid: "https://orcid.org/0000-0003-4925-7248"
cff-version: 1.2.0
date-released: "2021-07-18"
identifiers:
  - description: This is the collection of archived snapshots of all versions of My Research Software
    type: doi
    value: 10.5281/zenodo.123456
  - description: This is the archived snapshot of version 0.11.2 of My Research Software
    type: doi
    value: 10.5281/zenodo.123457
keywords:
  - "awesome software"
  - research
license: Apache-2.0
message: If you use this software, please cite it using these metadata.
repository-code: "https://github.com/citation-file-format/my-research-software"
title: My Research Software
version: 0.11.2
```
What's the logic behind which strings are quoted and which aren't? Some strings with spaces are quoted, some aren't. `date-released:` sometimes has them, sometimes not.

Is there a simple guide to which ones need to be quoted? Would it make sense to wrap *all* values in quotes in the examples?

### PR DESCRIPTION
**Pull request by sdruskat**
**Related issues**

Fixes #322

**Describe the changes made in this pull request**

- Lint examples for string quoting
- Explain the basics of YAML string quoting in first guide section
- Fix a few bad string examples (`"<string1>" <string2>`), which apparently weren't caught by linting

**Review checklist**

- [ ] Please run tests locally.

```bash
cd $(mktemp -d --tmpdir cff.XXXXXX)
git clone https://github.com/citation-file-format/citation-file-format .
git checkout <branch>
python3 -m venv env
source env/bin/activate
pip install --upgrade pip wheel setuptools
pip install -r requirements.txt
pytest
```

- [ ] Read the section on YAML string quoting and decide if it's suitable
- [ ] Quickly sanity check linting results

### CODE REVIEW

File: schema-guide.md [hunk 1]
```diff
@@ -404,26 +412,26 @@ primary principle, "Importance", when others cite this work.
...
+        date-released: "2021-07-26"
         doi: 10.5281/zenodo.x1234567
-        version: 0.13.4
         repository-code: "https://github.com/dependency-project/dependency"
-      - type: article
-        scope: Cite this paper if you want to reference the general concepts of the software.
-        authors:
+        title: Dependency
+        type: software
+        version: "0.13.4"
```
Comment (jspaaks) [inline schema-guide.md:422]: Do we treat these consistently? semver doesn't need to be quoted, but it's very close to being interpreted as a number

File: schema-guide.md [hunk 2]
```diff
@@ -959,16 +967,16 @@ authors:
...
-        date-end: 2018-07-27
-        date-start: 2021-07-27
-        email: "team@research-project.org"
+      - address: "742 Evergreen Terrace"
+        alias: RSP
+        city: Berlin
+        country: DE
+        date-end: "2018-07-27"
+        date-start: "2021-07-27"
+        email: team@research-project.org
```
Comment (jspaaks) [inline schema-guide.md:976]: shouldnt email addresses be quoted? yamllint says no, but your bulletlist at the top says yes

File: schema-guide.md [hunk 3]
```diff
@@ -5,8 +5,16 @@
 Valid Citation File Format files
 
 1. must be named `CITATION.cff` (note the capitalization);
-1. are valid YAML 1.2 ([specification](http://yaml.org/spec/1.2/spec.html), [validator](http://www.yamllint.com/));
-1. are valid according to the Citation File Format schema version 1.2.0 outlined in [schema.json](schema.json).
+1. are valid according to the Citation File Format schema version 1.2.0 outlined in [schema.json](schema.json);
+1. are valid YAML 1.2 ([specification](http://yaml.org/spec/1.2/spec.html), [validator](http://www.yamllint.com/)).
+
+<a name="yaml-strings"></a>**String quoting:** Note that in YAML you generally don't need to quote (`"<string>"`) strings.
```
Comment (jspaaks) [inline schema-guide.md:11]:
```suggestion
<a name="yaml-strings"></a>**String quoting:** Note that in YAML you generally don't need to quote strings.
```

File: schema-guide.md [hunk 4]
```diff
@@ -5,8 +5,16 @@
...
+But you should use `"` quotes when a string value
+contains whitespace,
+contains special characters (e.g., any of `:{}[],&*#?|-<>=!%@\`),
+consists only of numbers (e.g., is the string `"42"`, not the number `42`),
+or is `"true"`, `"false"`, `"yes"` or `"no"`.  
+In short: When a string value doesn't behave as expected, try putting it in `"` quotes.
```
Comment (jspaaks) [inline schema-guide.md:17]:
```suggestion
or is `"true"`, `"false"`, `"yes"` or `"no"`.
  
In short: When a string value doesn't behave as expected, try putting it in `"` quotes.
```

**Review by jspaaks — APPROVED** [review]
Nice work. I left a few suggestions for your consideration, but either way good to go

File: schema-guide.md (hunk 1 above)
Comment (sdruskat) [reply schema-guide.md:422]: We don't treat the consistently (yet). E.g., `cff-version` is always unquoted, `version` is mixed. 
Now that we have the text snippet about string quoting included, it should be safe to leave them all unquoted, although yes, they are close to number-like data. It's not something that we can solve, and I think it's better to have them in onw way or the other, and I think we decided against quoting everything?

Comment (sdruskat) [reply schema-guide.md]: Resolved in a997ed6 (see commit message for details)

File: schema-guide.md (hunk 2 above)
Comment (sdruskat) [reply schema-guide.md:977]: `@` is a reserved indicator in YAML and can't start a plain scalar.
So email addresses are safe to leave unquoted.
However, this should be addressed in the docs I guess, I'll see if I can do this in a non-distruptive way.

Comment (sdruskat) [reply schema-guide.md:977]: Resolved in cb83280.

**Comment by sdruskat** [comment]
> This is not my experience based on my interaction with GitHub (their Ruby Gem being awfully out of date notwithstanding).

I think it's not safe to use GitHub and their rendering as a source of truth for what can and cannot be done in YAML (and CFF).

> As I mentioned in my comment (in #322?), only colons have given me problems so far, and I’m only hypothesizing that hyphens might also be problematic if they start a string.

YAML has a number of reserved characters, including things like "reserved indicators", etc (`\`` and `@` for example), anchors use `&` and references `#`, `|` and `>` are used for specific types of multiline strings, brackets can generally be problematic as they signify different things in other formats that you would want to be able to convert to, etc. 

On top of this, I think the purpose of the schema guide snippet about quoting is not to give the maximally correct definition of when it is really necessary to quote special characters (we don't mention BOMs, hex values, etc. for that matter), but rather a safe and simple guideline that people can follow and which will cover a large percentage of real-world use cases.

**Comment by sdruskat** [comment]
> Nice work. I left a few suggestions for your consideration, but either way good to go

Thanks for the review, fixed some things and will merge now, and we can iterate if necessary.

**Pull request merged**